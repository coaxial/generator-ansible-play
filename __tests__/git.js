'use strict';
const gitRemoteOriginUrl = require('git-remote-origin-url');

const { readFileSync } = require('fs');
const assert = require('yeoman-assert');
const helpers = require('yeoman-test');
const path = require('path');

describe('generator-ansible-play:git', () => {
  describe('by default', () => {
    beforeAll(() => {
      return helpers.run(path.join(__dirname, '../generators/git')).withPrompts({
        initGit: true,
        gitHost: 'example.com',
        gitUser: 'coaxial',
        gitRepo: 'myrepo',
      });
    });

    it('initializes a git repo', () => {
      assert.file('.git');
    });

    it('generates a .gitignore file', () => {
      const actual = readFileSync('.gitignore', 'utf8');

      expect(actual).toMatchSnapshot();
    });

    it('sets the correct origin URL', () => {
      expect.assertions(1);

      const expected = 'git@example.com:coaxial/myrepo.git';

      return gitRemoteOriginUrl().then(originUrl => expect(originUrl).toEqual(expected));
    });
  });

  describe('when told not to init a git repo', () => {
    beforeAll(() => {
      return helpers
        .run(path.join(__dirname, '../generators/git'))
        .withPrompts({ initGit: false });
    });

    it('will not init a git repo', () => {
      assert.noFile('.git');
    });

    it('will not generate a .gitignore', () => {
      assert.noFile('.gitignore');
    });
  });
});
