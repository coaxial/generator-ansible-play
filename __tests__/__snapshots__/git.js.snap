// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator-ansible-play:git by default generates a .gitignore file 1`] = `
"*.key
*.log
*.pyc
*.retry
.vagrant
.vault_pass
__pycache__
.yo-rc.json
"
`;
